libyul/backends/evm/ControlFlowGraphBuilder.cpp

```diff
@@ -0,0 +1,484 @@
+			[](Scope::Variable&) { yulAssert(false, "Expected function name."); },
+			[&](Scope::Function& _function) { function = &_function; }
+		}), "Function name not found.");
+		yulAssert(function, "");
+		return m_currentBlock->operations.emplace_back(CFG::Operation{
+			// input
+			ranges::concat_view(
+				ranges::views::single(StackSlot{FunctionCallReturnLabelSlot{_call}}),
+				_call.arguments | ranges::views::reverse | ranges::views::transform(std::ref(*this))
+			) | ranges::to<Stack>,
+			// output
+			ranges::views::iota(0u, function->returns.size()) | ranges::views::transform([&](size_t _i) {
+				return TemporarySlot{_call, _i};
+			}) | ranges::to<Stack>,
+			// operation
+			CFG::FunctionCall{_call.debugData, *function, _call}
+		});
+	}
+}
+
+StackSlot ControlFlowGraphBuilder::operator()(FunctionCall const& _call)
+{
+	CFG::Operation& operation = visitFunctionCall(_call);
+	yulAssert(operation.output.size() == 1, "");
+	return operation.output.front();
+}
+
+void ControlFlowGraphBuilder::operator()(VariableDeclaration const& _varDecl)
+{
+	yulAssert(m_currentBlock, "");
+	auto declaredVariables = _varDecl.variables | ranges::views::transform([&](TypedName const& _var) {
+		return VariableSlot{lookupVariable(_var.name), _var.debugData};
+	}) | ranges::to<vector>;
+	Stack input;
+	if (_varDecl.value)
+		input = std::visit(util::GenericVisitor{
+			[&](FunctionCall const& _call) -> Stack {
+				CFG::Operation& operation = visitFunctionCall(_call);
+				yulAssert(declaredVariables.size() == operation.output.size(), "");
+				return operation.output;
+			},
+			[&](auto const& _identifierOrLiteral) -> Stack{
+				yulAssert(declaredVariables.size() == 1, "");
+				return {(*this)(_identifierOrLiteral)};
+			}
+		}, *_varDecl.value);
+	else
+		input = ranges::views::iota(0u, _varDecl.variables.size()) | ranges::views::transform([&](size_t) {
+			return LiteralSlot{0, _varDecl.debugData};
+		}) | ranges::to<Stack>;
+	m_currentBlock->operations.emplace_back(CFG::Operation{
+		std::move(input),
+		declaredVariables | ranges::to<Stack>,
+		CFG::Assignment{_varDecl.debugData, declaredVariables}
+	});
+}
+void ControlFlowGraphBuilder::operator()(Assignment const& _assignment)
+{
+	vector<VariableSlot> assignedVariables = _assignment.variableNames | ranges::views::transform([&](Identifier const& _var) {
+		return VariableSlot{lookupVariable(_var.name), _var.debugData};
+	}) | ranges::to<vector<VariableSlot>>;
+
+	yulAssert(m_currentBlock, "");
+	m_currentBlock->operations.emplace_back(CFG::Operation{
+		// input
+		std::visit(util::GenericVisitor{
+			[&](FunctionCall const& _call) -> Stack {
+				CFG::Operation& operation = visitFunctionCall(_call);
+				yulAssert(assignedVariables.size() == operation.output.size(), "");
+				return operation.output;
+			},
+			[&](auto const& _identifierOrLiteral) -> Stack {
+				yulAssert(assignedVariables.size() == 1, "");
+				return {(*this)(_identifierOrLiteral)};
+			}
+		}, *_assignment.value),
```

Comment:
This input code block is the same as in `operator()(vardecl)`. Maybe extract somehow?